packages/linter/src/rules/AmpImgUsesSrcSet.ts

```diff
@@ -0,0 +1,44 @@
+import { Context } from "../index";
+import { Rule } from "../rule";
+
+const CHECKED_IMG_LAYOUTS = ["fill", "flex-item", "intrinsic", "responsive"];
+
+const SVG_URL_PATTERN = /^[^?]+\.svg(\?.*)?$/i
+
+export class AmpImgUsesSrcSet extends Rule {
+  async run(context: Context) {
+    const $ = context.$;
+
+    const incorrectImages = $("amp-img")
+        .filter((_, e) => {
+          const src = $(e).attr("src");
+          let layout = $(e).attr("layout");
+          const srcset = $(e).attr("srcset");
+          const parent = $(e).parent();
+          if (parent.prop("tagName").startsWith('AMP-')) {
+            const parentLayout = $(parent).attr("layout");
+            if (parentLayout) {
+              layout = parentLayout;
+            }
+          }
+          return (
+            !SVG_URL_PATTERN.exec(src)
+            && layout && CHECKED_IMG_LAYOUTS.includes(layout)
+            && !srcset
+          )
+        });
+    if (incorrectImages.length > 0) {
+      return this.warn(
+        "Not all <amp-img> with non-fixed layout define a srcset. Using AMP Optimizer might help."
```

Comment:
Sorry I missed this the first time -- could be helpful to also provide which amp-img elements are affected, similar to how PSI is able to do: ![image](https://user-images.githubusercontent.com/10456171/94061779-d4d62a00-fdb3-11ea-92a9-79dd17cdf180.png)